### PR TITLE
Create_DbGender_DbUserGender

### PR DESCRIPTION
**Pull request by actimellka**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/939

### CODE REVIEW

File: src/UserService.Models.Db/DbGender.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    public const string TableName = "Genders";
+
+    public Guid Id { get; set; }
+    public string Name { get; set; }
+    public Guid CreatedBy { get; set; }
+    public DateTime CreatedAtUtc { get; set; }
+    public ICollection<DbUserGender> UserGenders { get; set; }
+  }
+
+  public class GenderConfiguretion : IEntityTypeConfiguration<DbGender>
```
Comment (irlirla) [inline src/UserService.Models.Db/DbGender.cs:19]: Please, change to GenderConfiguration

**Review by irlirla — APPROVED** [review]

File: src/UserService.Models.Db/DbUserGender.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using System;
+
+namespace LT.DigitalOffice.UserService.Models.Db
+{
+  public class DbUserGender
+  {
+    public const string TableName = "UserGenders";
```
Comment (Cheephot) [inline src/UserService.Models.Db/DbUserGender.cs:9]: UsersGenders

Comment (Cheephot) [reply src/UserService.Models.Db/DbUserGender.cs:9]: Accordingly, do not forget to change it everywhere

Comment (actimellka) [reply src/UserService.Models.Db/DbUserGender.cs]: done

File: src/UserService.Models.Db/DbGender.cs (hunk 1 above)
Comment (actimellka) [reply src/UserService.Models.Db/DbGender.cs]: done

File: src/UserService.Data.Provider.MsSql.Ef/UserServiceDbContext.cs [hunk 3]
```diff
@@ -24,7 +24,9 @@ public class UserServiceDbContext : DbContext, IDataProvider
     public DbSet<DbUserEducation> UserEducations { get; set; }
     public DbSet<DbPendingUser> PendingUsers { get; set; }
     public DbSet<DbEntityImage> EntitiesImages { get; set; }
-    public DbSet<DbUserLocation> UsersLocations { get; set; }
+    public DbSet<DbUserLocation> UsersLocations { get; set; }
+    public DbSet<DbUserGender> UserGenders { get; set; }
```
Comment (MariyaSin) [inline src/UserService.Data.Provider.MsSql.Ef/UserServiceDbContext.cs:28]: UsersGenders

File: src/UserService.Models.Db/DbGender.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+{
+  public class DbGender
+  {
+    public const string TableName = "Genders";
+
+    public Guid Id { get; set; }
+    public string Name { get; set; }
+    public Guid CreatedBy { get; set; }
+    public DateTime CreatedAtUtc { get; set; }
+    public ICollection<DbUserGender> UserGenders { get; set; }
```
Comment (MariyaSin) [inline src/UserService.Models.Db/DbGender.cs:16]: UsersGenders

File: src/UserService.Models.Db/DbUser.cs [hunk 5]
```diff
@@ -43,6 +43,7 @@ public class DbUser
     public ICollection<DbUserSkill> Skills { get; set; }
     [IgnoreParse]
     public DbUserLocation Location { get; set; }
+    public DbUserGender Genders { get; set; }
```
Comment (MariyaSin) [inline src/UserService.Models.Db/DbUser.cs:46]: Gender

File: src/UserService.Models.Db/DbUserGender.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      builder
+        .HasKey(ug => ug.Id);
+
+      builder
+        .Property(ug => ug.UserId)
+        .IsRequired();
+
+      builder
+        .Property(ug => ug.GenderId)
+        .IsRequired();
```
Comment (MariyaSin) [inline src/UserService.Models.Db/DbUserGender.cs:36]: no need

**Review by grinakate — APPROVED** [review]

**Review by VladSpitsin — APPROVED** [review]

**Review by asyagraf — APPROVED** [review]

File: src/UserService.Models.Db/DbUserGender.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    {
+      builder
+        .ToTable(DbUserGender.TableName);
+
+      builder
+        .HasKey(ug => ug.Id);
+
+      builder
+        .Property(ug => ug.UserId)
+        .IsRequired();
```
Comment (MariyaSin) [inline src/UserService.Models.Db/DbUserGender.cs:32]: no need write it

File: src/UserService.Models.Db/DbUserGender.cs [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
```
Comment (MariyaSin) [inline src/UserService.Models.Db/DbUserGender.cs:33]: remove empty line

**Review by MariyaSin — APPROVED** [review]

**Pull request merged**